http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HeadersIT.java

```diff
@@ -93,6 +96,23 @@ void testPathSpecificHeaderRulesOrder() {
         cacheControlMatches(response, "max-age=1");
     }
 
+    /*
+     * This test check 3.8.6 backport https://github.com/quarkusio/quarkus/pull/41411 will fails with quarkus-resteasy
```

Comment:
Can you add tag `@Tag("https://github.com/quarkusio/quarkus/pull/41411")`

Also I think `HeadersMessageBodyWriter` should work with both or at least I would create issue if it's not work.

As reading this test it's work with that specific header but not with `MediaType.TEXT_PLAIN`. So I would say it's a bug